front/app/components/admin/FormResults/FormResultsQuestion/TextQuestion/Analysis/utils.ts

```diff
@@ -13,3 +23,92 @@ export const convertFilterValuesToString = (filters?: IInputsFilterParams) => {
     }, {})
   );
 };
+
+type FilterForQuarterArgs = {
+  insights?: IInsights;
+  search: URLSearchParams;
+  analysisSummaries?: UseQueryResult<ISummary, unknown>[];
+};
+
+/**
+ * Filters insights to only include those published within the selected quarter (from URL params).
```

Comment:
I'll add unit tests for these functions on Monday :) But code is ready to review I think already.